src/Engine/EmitMSIL/CodeGenIL.cs

```diff
@@ -1286,6 +1288,15 @@ private Type EmitFunctionCallNode(AssociativeNode node)
                 {
                     EmitOpCode(OpCodes.Box, t);
                 }
+
+                EmitOpCode(OpCodes.Ldc_I4, 3);
+
+                // Emit opcodes to create a new CLRStackValue structure
+                // with value and proto core type set in the constructor
+                var constr = typeof(DSASM.CLRStackValue).GetConstructor(new Type[] { typeof(object), typeof(int) });
+                EmitOpCode(OpCodes.Newobj, constr); 
+
+                EmitOpCode(OpCodes.Box, typeof(DSASM.CLRStackValue));
```

Comment:
@mjkkirschner this was just a test.
I think the proper solution will be to do conversion to CLRStackValue or something....

Reply:
👍